jni/patches/nmslib/0002-Adds-ability-to-pass-ef-parameter-in-the-query-for-h.patch

```diff
@@ -0,0 +1,303 @@
+From 22c609af609e6f4214e67df0c405d2c86014c403 Mon Sep 17 00:00:00 2001
+From: Tejas Shah <shatejas@amazon.com>
+Date: Mon, 27 May 2024 22:02:12 -0700
+Subject: [PATCH] Adds ability to pass ef parameter in the query for hnsw
+
+It defaults to index ef_ value if its not type HNSWQuery
```

Comment:
Here is the [diff](https://github.com/shatejas/nmslib/commit/22c609af609e6f4214e67df0c405d2c86014c403) for this patch

Reply:
There are no unit test in the entire lib. There is no way to test it out unless we end up writing it for the entire search which will become a heavy lift.

There is an integ test though and I tested it out using this code https://github.com/shatejas/nmslib/commit/c5dd7cdf6d3c16d3646a434e1d6a7546a9a3a7fc. I did not add it in the patch because its not generic enough. There are some validations in the hnsw code which is making it difficult to add more parameters in the integ test

@heemin32